### DISCUSSION TITLE
Adding LetsEncrypt's Root Certificate

### DISCUSSION THREAD

**Discussion by l008com** [Homebrew/discussions#2300] [Everyday usage]
I'm running Homebrew on an older OS. When LetsEncrypt's root certificate expired, my OS couldn't connect to LetsEncrypt SLL domains any more. It's an easy enough fix though, I just added their root cert to the keychain app.

However it appears (though I'm not 100% certain how it works) that PHP installed via homebrew is using a separate set of root certs, I'm guessing ones included in openssl also installed by homebrew? 

Is this accurate? I'm kind of half speculating here.

If that is the case, how do I manually add their root cert? I have a bunch of php shell scripts that do some data grabbing and processing, and since the world all but made everyone start using LetsEncrypt, suddenly lots of sites are unreachable. 

I know upgrading to a newer server with a newer OS would be the best solution to this problem, but that's really not an option right now, for many reasons. 

Here's some of the kind of errors I get when trying to connect to LetsEncrypt domains from within my PHP scripts:

> PHP Warning:  stream_socket_client(): SSL operation failed with code 1. OpenSSL Error messages:
> error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed in /scripts/ssl.test.php on line 9
> 
> Warning: stream_socket_client(): SSL operation failed with code 1. OpenSSL Error messages:
> error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed in /scripts/ssl.test.php on line 9
> PHP Warning:  stream_socket_client(): Failed to enable crypto in /scripts/ssl.test.php on line 9

**Comment by SMillerDev** [comment]
If you run a `brew upgrade` it should update openssl to use the newest certificates.

**Reply by l008com** [reply]
It can't update openssl because - you guess it - it can't verify any of the certificates to download any packages :D

**Reply by SMillerDev** [reply]
Try doing `brew update` first because we've been adding plain HTTP mirrors for this reason.

**Reply by l008com** [reply]
Ok doing the update, and the git command first does let it download packages again. Now I'm just being hung up by the same problem i've had before, that enough packages won't install on el capitan that I can't really update anything. Total bummer.
